vms/platformvm/txs/mempool/mempool.go

```diff
@@ -224,17 +224,12 @@ func (m *mempool) HasTxs() bool {
 }
 
 func (m *mempool) PeekTxs(maxTxsBytes int) []*txs.Tx {
-	txs := m.unissuedDecisionTxs.List()
-	txs = append(txs, m.unissuedStakerTxs.List()...)
-
-	size := 0
-	for i, tx := range txs {
-		size += len(tx.Bytes())
-		if size > maxTxsBytes {
-			return txs[:i]
-		}
+	txs, remaining := m.unissuedDecisionTxs.ListWithLimit(maxTxsBytes)
```

Comment:
nice cleanup. I'd go one step further and remove the List method completely in favour of the ListWithLimit one. We can pass MaxInt in case we want the full txs list and we'd keep the heap interface simpler.